src/System.Management.Automation/namespaces/ProviderBase.cs

```diff
@@ -64,6 +64,8 @@ public abstract partial class CmdletProvider : IResourceSupplier
         /// </summary>
         private ProviderInfo _providerInformation = null;
 
+        private readonly PSObjectBuilderUnsafe _psObjectBuilderUnsafe = new PSObjectBuilderUnsafe(10);
```

Comment:
Why should we use unsafe version here?

Reply:
The PSObjectBuilder cannot leave a stack frame. It cannot be a member of a type or allocated on the heap.

So the less optimized version has to be used if it is to survive between calls.

Reply:
And again - Think of PSObjectBuilderUnsafe as PSObjectBuilderLessOptimized.

Reply:
I will rename it. I see that it causes confusion from the start.

Reply:
I like that suggestion! Changing!